src/auto-advance.js

```diff
@@ -0,0 +1,58 @@
+import window from 'global/window';
+
+/**
+ * Validates a number of seconds to use as the auto-advance delay.
+ *
+ * @param  {Number} s
+ * @return {Boolean}
+ */
+const validSeconds = s =>
+  typeof s === 'number' && !isNaN(s) && s >= 0 && s < Infinity;
+
+/**
+ * Resets the auto-advance behavior of a player.
+ *
+ * @param  {Player} player
+ */
+const reset = (player) => {
+  if (player.playlist.autoadvance_.timeout) {
+    window.clearTimeout(player.playlist.autoadvance_.timeout);
+  }
+
+  if (player.playlist.autoadvance_.trigger) {
+    player.off('ended', player.playlist.autoadvance_.trigger);
+  }
+
+  player.playlist.autoadvance_.timeout = null;
+  player.playlist.autoadvance_.trigger = null;
+};
+
+/**
+ * Sets up auto-advance behavior on a player.
+ *
+ * @param  {Player} player
+ * @param  {Number} seconds
```

Comment:
Though, I still think `timeout` might be a more appropriate variable name because it is the auto-advance *timeout*.